### DISCUSSION TITLE
MOOSE THM Brayton Cycle example: missing `thm_utilities` Python module

### DISCUSSION THREAD

**Discussion by oleachri** [idaholab/moose#27473] [Q&A Modules: Thermal Hydraulics] [answered]
I've been investigating the MOOSE THM Brayton Cycle examples.  I can get these to run fine, but I noticed the distribution includes a Python script, `plot.py` i.e.:

[modules/thermal_hydraulics/test/tests/problems/brayton_cycle/plot.py](https://github.com/idaholab/moose/blob/2fd248e338982ab5b2dc53548af01210c5aa7ace/modules/thermal_hydraulics/test/tests/problems/brayton_cycle/plot.py)

... which has a line `from thm_utilities import readCSVFile`.  However, my MOOSE install doesn't have the Python module `thm_utilities`.  Any tips on how I can obtain and install it?

**Comment by GiudGiud** [comment, marked as answer]
Hello

That script is found in moose//modules/thermal_hydraulics/python/
You can use "find" on linux to find a missing file in a repo

Guillaume

**Reply by oleachri** [reply]
Thanks for the quick reply Guillaume.  I should probably have stated that I'm using the pre-built MOOSE binaries, `snapshot-20-10-27-34878-gf2bb1b9505`, and `thm_utilities` doesn't appear to ship with it (I have checked using `find`).  I have obtained the file manually from the location you mention, in the [online repo](https://github.com/idaholab/moose/blob/2fd248e338982ab5b2dc53548af01210c5aa7ace/modules/thermal_hydraulics/python/thm_utilities.py).

This seems to work, though it falls over at a later stage with a LaTeX error that's probably due to our system config.:

```
raise RuntimeError(
RuntimeError: latex was not able to process the following string:
b'lp'

Here is the full report generated by latex:
This is pdfTeX, Version 3.1415926-2.5-1.40.14 (TeX Live 2013/TeX Live for SUSE Linux)
 restricted \write18 enabled.
entering extended mode
(~/.cache/matplotlib/tex.cache/1acea6f6c115d0ec7a634ed0529287b9.tex
LaTeX2e <2011/06/27>
Babel <3.9f> and hyphenation patterns for 78 languages loaded.
(/usr/share/texmf/tex/latex/base/article.cls
Document Class: article 2007/10/19 v1.4h Standard LaTeX document class
(/usr/share/texmf/tex/latex/base/size10.clo))

! LaTeX Error: File `type1cm.sty' not found.

Type X to quit or <RETURN> to proceed,
or enter new name. (Default extension: sty)

Enter file name: 
! Emergency stop.
<read *> 
         
l.5 \usepackage
               {type1ec}^^M
No pages of output.
Transcript written on 1acea6f6c115d0ec7a634ed0529287b9.log.
```

**Reply by GiudGiud** [reply]
hello

this does look like a latex error, a missing template file maybe
it seems you just need to install dvipng
https://stackoverflow.com/questions/11354149/python-unable-to-render-tex-in-matplotlib

**Reply by oleachri** [reply]
Great - thanks for your help.
